src/isolate-duplicates/index.js

```diff
@@ -1,3 +1,54 @@
-function isolateDuplicates(text) {}
+function isolateDuplicates(text) {
+    if (typeof(text) != "string"){
+        throw "Please enter a valid string";
+    }
+    else{
+        let splittedString = text.split('');
+        let frequency = 0;
+        let insertions = 0;
+        for (let i = 0; i < splittedString.length; i++){
+            let letter1 = splittedString[i];
+            let letter2 = splittedString[i + 1];
+            const string1 = convert(letter1);
+            const string2 = convert(letter2);
+            if (string1 === string2){
+                frequency++;
+                if (frequency == 2){
+                    splittedString.splice(i + 1, 0, '[');
+                    frequency++;
+                    i++;
+                    insertions++
+                }
+                
+            } 
+            else if (frequency > 2 ){
+                if (string1 != string2){
+                    splittedString.splice(i + 1, 0, ']');
+                    i++;
+                    insertions++
+                    frequency = 0;
+                }
+            }
+            else {
+                frequency = 0;
+            }
+        }
+        let result = [splittedString.join(''), insertions /2]
+        return result;
+    }
+}
+
+function convert(str){
+    let uppercase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    let lowercase = 'abcdefghijklmnopqrstuvwxyz';
+    for (let i = 0; i < uppercase.length; i++){
+        if (uppercase[i] == str){
+            return lowercase[i];
+        }
+        else if (lowercase[i] == str){
+            return str;
+        }
+    }
```

Comment:
Why not just use `str.toLowerCase()`. This seems redundant.

Reply:
I tried using str.toLowerCase() but encountered a problem. 

<img width="438" alt="toLowerCase()" src="https://user-images.githubusercontent.com/55290410/80861675-f8170600-8c67-11ea-83cd-c20ae139884e.png">

I was trying to figure out why it wasn't working but I wasn't having much time to complete the challenge. So I wrote an alternative code to solve it. I will appreciate if @bondz  or anyone can tell me why toLowerCase() was not working. Thank you, sir.

Reply:
Assume the string in question is `brave`, at the last iteration, the index is 4 for the letter `e`, but this is what your code looks like...

```js
const letter2 = splittedString(i + 1) // 4 + 1 -> 5
// There's no letter at the 5th index, so `letter2` is undefined.
...
const string2 = letter2.toLowerCase() // essentially, undefined.toLowerCase
```

And this is why you get the classic error message, cannot read property "..." of undefined, because the variable you're calling that property on is undefined.

This is why you want to avoid reading beyond index ranges this way as you'd have to guard a lot to make sure you're not reading out of range.

Reply:
Thank you so much @bondz. Now I understand. I have been able to implement it.